### DISCUSSION TITLE
How quickly are new kubernetes pod labels detected?

### DISCUSSION THREAD

**Discussion by airhorns** [vectordotdev/vector#7800] [Q&A] [answered]
Hi folks -- thanks for such a wonderful piece of open source software?

I am using the `kubernetes_logs` source and processing the log entries just fine, but I have some fancy stuff running in kubernetes that applies new pod labels to existing pods, long after they started. Vector picks up these new labels in the autodiscovered data it sticks in each event eventually, but I am struggling to discern when or if/when there is a delay. How fast should we expect new pod labels to be reflected in emitted log events? Is there a way to control this timing?

Thanks for any insight you can share!

**Comment by spencergilbert** [comment, marked as answer]
It should be relatively _real time_ we have a watcher on pod events so hand waving some questions about distributed systems, vector will receive updates pretty quickly from the control plane on any changes. That said I don't think we've done any measurements here.

**Reply by airhorns** [reply]
Hey that's awesome! Ok! Bless your heart!
